book/db/README.md

```diff
@@ -0,0 +1,332 @@
+Lets look at an examples of how cursors are used. The code snippet below is the `unwind` method from the `BodyStage` defined in the `stages` crate. This function is responsible for unwinding any changes to the database if there is an error when executing the body stage within the Reth pipeline. 
+
+[File: crates/stages/src/stages/bodies.rs](https://github.com/paradigmxyz/reth/blob/main/crates/stages/src/stages/bodies.rs#L205-L238)
+```rust ignore
+ /// Unwind the stage.
+    async fn unwind(
+        &mut self,
+        db: &mut Transaction<'_, DB>,
+        input: UnwindInput,
+    ) -> Result<UnwindOutput, Box<dyn std::error::Error + Send + Sync>> {
+        let mut tx_count_cursor = db.cursor_mut::<tables::CumulativeTxCount>()?;
+        let mut block_ommers_cursor = db.cursor_mut::<tables::BlockOmmers>()?;
+        let mut transaction_cursor = db.cursor_mut::<tables::Transactions>()?;
+
+        let mut entry = tx_count_cursor.last()?;
+        while let Some((key, count)) = entry {
+            if key.number() <= input.unwind_to {
+                break
+            }
+
+            tx_count_cursor.delete_current()?;
+            entry = tx_count_cursor.prev()?;
+
+            if block_ommers_cursor.seek_exact(key)?.is_some() {
+                block_ommers_cursor.delete_current()?;
+            }
+
+            let prev_count = entry.map(|(_, v)| v).unwrap_or_default();
+            for tx_id in prev_count..count {
+                if transaction_cursor.seek_exact(tx_id)?.is_some() {
+                    transaction_cursor.delete_current()?;
+                }
+            }
+        }
+
+    //--snip--
+    }
+
+```
+
+This function first grabs a mutable cursor for the `CumulativeTxCount`, `BlockOmmers` and `Transactions` tables. 
+
+The `tx_count_cursor` is used to get the last key value pair written to the `CumulativeTxCount` table and delete key value pair where the cursor is currently pointing.
+
+The `block_ommers_cursor` is used to get the block ommers from the `BlockOmmers` table at the specified key, and delete the entry where the cursor is currently pointing.
+
+Finally, the `transaction_cursor` is used to get delete each transaction from the last `TXNumber` written to the database, to the current tx count.
+
+While this is a brief look at how cursors work in the context of database tables, the chapter on the `libmdbx` crate will go into further detail on how cursors communicate with the database and what is actually happening under the hood.
+
+<br>
+
+## Summary
+This chapter was packed with information, so lets do a quick review. The database is comprised of tables, with each table being a collection of key-value pairs representing various pieces of data for each block in the blockchain. Any struct that implements the `Database` trait can view, update or delete entries in the various tables. The database design leverages nested traits and generic associated types to provide methods to interact with each table in the database.
```

Comment:
"representing various pieces of data for each block in the blockchain" feels a bit too specific, a reader might get confused and think this means the tables only store blocks - maybe just say "in the blockchain" instead of "for each block in the blockchain"?